elf_reader.go

```diff
@@ -32,6 +33,18 @@ type kfuncMeta struct {
 	Func    *btf.Func
 }
 
+type ksymMetaKey struct{}
+
+type ksymMeta struct {
+	Binding elf.SymBind
+	Var     *ksymVarSpec
+}
+
+type ksymVarSpec struct {
```

Comment:
I'd do away with this type altogether, I'm not sure what value it adds. elfCode only needs to memoize the symbol name and offset.

```go
type ksymMeta struct {
	Binding elf.SymBind
	Addr    uint64
	Name    string
}

type elfCode struct {
	...
	kfuncs   map[string]*btf.Func
	ksyms    map[string]uint64
	kconfig  *MapSpec
}
```

In loadKsymsSection():
```go
       var ksyms []string
        for _, v := range ds.Vars {
                switch t := v.Type.(type) {
...
                case *btf.Var:
                       ec.ksymVars[t.TypeName()] = &ksymVarSpec{
                               Var: t,
                       }
                       ksyms = append(ksyms, t.TypeName())
...

       ksymAddrs, err := kallsyms.LoadSymbolAddresses(ksyms)
...
```